File: src/hp_guard/logging.py
from __future__ import annotations
from datetime import datetime, timezone
from .models import AuditEntry, PolicyCall
from .engine import Decision


class AuditLogger:
    def log(self, call: PolicyCall, decision: Decision) -> AuditEntry:
        return AuditEntry(
            timestamp=datetime.now(timezone.utc).isoformat(),
            agent=call.agent,
            tool=call.tool,
            args=list(call.args),
            decision=decision.action,
            matched_rules=list(decision.matched_rules),
        )

File: src/logging.rs
use crate::models::{AuditEntry, Decision, PolicyCall};
use chrono::Utc;

/// Logs audit entries for policy decisions.
pub struct AuditLogger;

impl AuditLogger {
    pub fn log(&self, call: &PolicyCall, decision: &Decision) -> AuditEntry {
        AuditEntry {
            timestamp: Utc::now(),
            agent: call.agent.clone(),
            tool: call.tool.clone(),
            args: call.args.clone(),
            decision: decision.action,
            matched_rules: decision.matched_rules.clone(),
        }
    }
}
